### PR TITLE
LUCENE-9765: Hunspell: rename SpellChecker to Hunspell, fix test name…

### PR DESCRIPTION
**Pull request by donnerpeter**
…, update javadoc and CHANGES.txt

# Description

The class names are imperfect and the docs are outdated

# Solution

Fix that

# Tests

No behavior change.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -89,8 +89,8 @@ API Changes
 
 Improvements
 
-* LUCENE-9687: Hunspell support improvements: add SpellChecker API, support default encoding and
-  BREAK/FORBIDDENWORD/COMPOUNDRULE affix rules, improve stemming of all-caps words (Peter Gromov)
+* LUCENE-9687: Hunspell support improvements: add API for spell-checking and suggestions, support compound words,
+  fix various behavior differences between Java and C++ implementations, improve performance (Peter Gromov, Dawid Weiss)
```
Comment (dweiss) [inline lucene/CHANGES.txt:93]: I don't think I deserve to be in the changes entry. :)

Comment (donnerpeter) [reply lucene/CHANGES.txt:93]: I think you do, as you've dedicated quite a lot of your time to my and our own PRs. We could also change that to "mostly Peter Gromov, but also Dawid Weiss" if you like :)

Comment (dweiss) [reply lucene/CHANGES.txt:93]: There's actually a convention for this: "Peter Gromov via Dawid Weiss".

Comment (dweiss) [reply lucene/CHANGES.txt:93]: Doesn't matter.

Comment (donnerpeter) [reply lucene/CHANGES.txt:93]: I thought "via" is for pure review activity, but you've made meaningful changes yourself which I didn't even plan. Feel free to change to "via" if you like :)

**Pull request merged**